semgrep_output_v1.atd

```diff
@@ -396,37 +394,60 @@ type svalue_value <python decorator="dataclass(frozen=True)"> = {
 (*****************************************************************************)
 (* Match dataflow trace *)
 (*****************************************************************************)
-
 (* EXPERIMENTAL *)
+
+(* It's easier to understand the dataflow trace data structures on a simple
+ * example. Here is one simple Python target file:
+ *
+ * 1:   def foo():
+ * 2:     return source()
+ * 3:   
+ * 4:   def bar(v):
+ * 5:     sink(v)
+ * 6:   
+ * 7:   x = foo()
+ * 8:   y = x
+ * 9:   bar(y)
+ *
+ * and here is roughly the generated match_dataflow_trace assuming
+ * a Semgrep rule where source() is a taint source and sink() the taint sink:
+ *
+ *  taint_source = CliCall("foo() @l7", [], CliLoc "source() @l2")
+ *  intermediate_vars = ["x", "y"]
+ *  taint_sink = CliCall("bar()" @l9, ["v"], CliLoc "sink(v) @l5")
+ *)
+
 type match_dataflow_trace <python decorator="dataclass(frozen=True)"> = {
   ?taint_source: match_call_trace option;
-  (* Intermediate variables which are involved in the dataflow. For taint, this
+  (* Intermediate variables which are involved in the dataflow. This
    * explains how the taint flows from the source to the sink. *)
   ?intermediate_vars: match_intermediate_var list option;
-  (* For now, omitting the taint sink, since it's redundant data except for
-   * certain cases that only appear with DeepSemgrep. We can add it in later
-   * without breaking consumers. *)
   ?taint_sink: match_call_trace option;
 }
 
-(* the string attached to the location is the actual text from the file
- * TODO: define a location_with_content type that can be reused.
+(* The string attached to the location is the actual code from the file.
+ * This can contain sensitive information so be careful!
+ * 
+ * TODO: the type seems redundant since location already specifies a range.
+ * maybe this saves some effort to the user of this type which do not
+ * need to read the file to get the content.
  *)
+type loc_and_content = (location * string)
+
 type match_call_trace <python decorator="dataclass(frozen=True, order=True)"> = [
-  | CliLoc of (location * string)
-  | CliCall of ((location * string) * match_intermediate_var list * match_call_trace)
+  | CliLoc of loc_and_content
+  | CliCall of (loc_and_content * match_intermediate_var list * match_call_trace)
 ] <ocaml repr="classic">
 
 
-(* EXPERIMENTAL *)
+(* This type happens to be mostly the same as a loc_and_content for now, but
+ * it's split out because Iago has plans to extend this with more information
+*)
 type match_intermediate_var <python decorator="dataclass(frozen=True)"> = {
```

Comment:
Also, this right now is guaranteed to refer to a variable/l-value, whereas `loc_and_content` could be an arbitrary thing.